src/renderer/atlas/BackendD3D.cpp

```diff
@@ -295,20 +295,20 @@ void BackendD3D::_updateFontDependents(const RenderingPayload& p)
     // baseline of curlyline is at the middle of singly underline. When there's
     // limited space to draw a curlyline, we apply a limit on the peak height.
     {
-        const auto cellHeight = static_cast<f32>(font.cellSize.y);
-        const auto duTop = static_cast<f32>(font.doubleUnderline[0].position);
-        const auto duBottom = static_cast<f32>(font.doubleUnderline[1].position);
-        const auto duHeight = static_cast<f32>(font.doubleUnderline[0].height);
+        const int cellHeight = font.cellSize.y;
+        const int duTop = font.doubleUnderline[0].position;
+        const int duBottom = font.doubleUnderline[1].position;
+        const int duHeight = font.doubleUnderline[0].height;
 
         // This gives it the same position and height as our double-underline. There's no particular reason for that, apart from
         // it being simple to implement and robust against more peculiar fonts with unusually large/small descenders, etc.
-        // We still need to ensure though that it doesn't clip out of the cellHeight at the bottom.
-        const auto height = std::max(3.0f, duBottom + duHeight - duTop);
-        const auto top = std::min(duTop, floorf(cellHeight - height - duHeight));
+        // We still need to ensure though that it doesn't clip out of the cellHeight at the bottom, which is why `position` has a min().
+        const auto height = std::max(3, duBottom + duHeight - duTop);
+        const auto position = std::min(duTop, cellHeight - height - duHeight);
 
         _curlyLineHalfHeight = height * 0.5f;
-        _curlyUnderline.position = gsl::narrow_cast<u16>(lrintf(top));
-        _curlyUnderline.height = gsl::narrow_cast<u16>(lrintf(height));
+        _curlyUnderline.position = gsl::narrow_cast<u16>(position);

```

Comment:
no more rounding - all this math now uses truncation. is that OK?

Reply:
(I don't 100% know why the changes to the D3D backend were needed -- can you explain? Did you test them again at different font sizes and scalings?)

Reply:
I did this because I copied the code over to D2D. While doing so I noticed that this code doesn't actually do any floating point math whatsoever. As you can see I don't need to do any explicit integer/float casts anywhere in the preceding lines (which would normally be required during AuditMode). I changed the D3D part so that it's mostly identical to the D2D part.